src/librustc_metadata/decoder.rs

```diff
@@ -1057,67 +1125,66 @@ impl<'a, 'tcx> CrateMetadata {
+                let reusable_filemap = {
+                    local_codemap.files
+                        .borrow()
+                        .iter()
+                        .find(|fm| are_equal_modulo_startpos(&fm, &filemap_to_import))
+                        .map(|rc| rc.clone())
+                };
 
-            match reusable_filemap {
-                Some(fm) => {
-                    cstore::ImportedFileMap {
-                        original_start_pos: filemap_to_import.start_pos,
-                        original_end_pos: filemap_to_import.end_pos,
-                        translated_filemap: fm
-                    }
-                }
-                None => {
-                    // We can't reuse an existing FileMap, so allocate a new one
-                    // containing the information we need.
-                    let syntax_pos::FileMap {
-                        name,
-                        abs_path,
-                        start_pos,
-                        end_pos,
-                        lines,
-                        multibyte_chars,
-                        ..
-                    } = filemap_to_import;
-
-                    let source_length = (end_pos - start_pos).to_usize();
-
-                    // Translate line-start positions and multibyte character
-                    // position into frame of reference local to file.
-                    // `CodeMap::new_imported_filemap()` will then translate those
-                    // coordinates to their new global frame of reference when the
-                    // offset of the FileMap is known.
-                    let mut lines = lines.into_inner();
-                    for pos in &mut lines {
-                        *pos = *pos - start_pos;
-                    }
-                    let mut multibyte_chars = multibyte_chars.into_inner();
-                    for mbc in &mut multibyte_chars {
-                        mbc.pos = mbc.pos - start_pos;
+                match reusable_filemap {
+                    Some(fm) => {
+                        cstore::ImportedFileMap {
+                            original_start_pos: filemap_to_import.start_pos,
+                            original_end_pos: filemap_to_import.end_pos,
+                            translated_filemap: fm,
+                        }
                     }
+                    None => {
+                        // We can't reuse an existing FileMap, so allocate a new one
+                        // containing the information we need.
+                        let syntax_pos::FileMap { name,
+                                                  abs_path,
+                                                  start_pos,
+                                                  end_pos,
+                                                  lines,
+                                                  multibyte_chars,
+                                                  .. } = filemap_to_import;
```

Comment:
IMO this looked much nicer before with block indenting and a newline after the `{`.